src/selectors/getBreakpoints.js

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
+
+// @flow
+
+import { createSelector } from "reselect";
+
+import { makeLocationId } from "../utils/breakpoint";
+import { isInterrupted } from "../utils/pause";
+
+import {
+  getSources,
+  getSourceInSources,
+  getBreakpoints,
+  getPauseReason,
+  getTopFrame
+} from "../selectors";
+
+export const getBreakpointSelector = createSelector(
+  getBreakpoints,
+  getSources,
+  getTopFrame,
+  getPauseReason,
+  (breakpoints, sources, frame, why) =>
+    breakpoints
+      .map(bp => updateLocation(sources, frame, why, bp))
+      .filter(
+        bp => bp.location.source && !bp.location.source.get("isBlackBoxed")
+      )
+);
+
+function updateLocation(sources, frame, why, bp) {
+  const source = getSourceInSources(sources, bp.location.sourceId);
+  const isCurrentlyPaused = isCurrentlyPausedAtBreakpoint(frame, why, bp);
+  const locationId = makeLocationId(bp.location);
+
+  const location = { ...bp.location, source };
+  const localBP = { ...bp, location, locationId, isCurrentlyPaused };
+
+  return localBP;
+}
+
+function isCurrentlyPausedAtBreakpoint(frame, why, breakpoint) {
+  if (!frame || !isInterrupted(why)) {
+    return false;
+  }
+
+  const bpId = makeLocationId(breakpoint.location);
+  const pausedId = makeLocationId(frame.location);
+  return bpId === pausedId;
+}
```

Comment:
Is this file going to contain all the selectors for breakpoints or is it one file per selector? if it is the former , then this file may probably be  "src/selectors/breakpoints" rather than "src/selectors/getBreakpoints"?

Reply:
yeah - getBreakpointsForPane

Reply:
hmm, i think i misread this comment... 

I thought you were referring to the selector function for the breakpoints component. Not sure about the reducers.